scripts/docgen/snippets/extractSnippets.ts

```diff
@@ -44,96 +46,50 @@ export const extractPackageSnippets = ({
-    applyTransforms(sourceFile, ctx)
-    const snippetRanges = extractSnippetRangesFromFile(sourceFile)
-    const byLabel = extractTextFromSnippetRanges(sourceFile, snippetRanges)
+const extractSnippetsFromFile = (sourceFileText: string): FileSnippets => {
+    const byLabel = extractLabeledSnippets(sourceFileText)
+    const text = filterSnipStatements(sourceFileText.split("\n"))
     return {
         all: {
-            // It's important we get the file text after extracting snippets so snip comments are not included
-            text: sourceFile.getFullText().trim()
+            text
         },
         byLabel
     }
 }
 
-const applyTransforms = (
-    sourceFile: SourceFile,
-    ctx: ExtractFileSnippetContext
-) => {
-    if (ctx.transforms.imports) {
-        // Replace relative internal imports with standard external imports
-        const importDeclarations = sourceFile.getDescendantsOfKind(
-            SyntaxKind.ImportDeclaration
-        )
-        for (const declaration of importDeclarations) {
-            const specifier = declaration.getModuleSpecifier()
-            if (specifier.getLiteralText().endsWith("src/index.js")) {
-                specifier.replaceWithText(`"${ctx.packageMetadata.name}"`)
-            }
-        }
-    }
-}
+export type LabeledSnippets = Record<string, Snippet>
 
-const extractTextFromSnippetRanges = (
-    sourceFile: SourceFile,
-    snippetRanges: SnipRange[]
-): LabeledSnippets => {
+const extractLabeledSnippets = (sourceFileText: string): LabeledSnippets => {
     const labeledSnippets: Record<string, Snippet> = {}
-    for (const resolvedSnip of snippetRanges) {
-        let text
-        if ("node" in resolvedSnip.range) {
-            text = resolvedSnip.range.node.getFullText()
-        } else {
-            text = sourceFile
-                .getFullText()
-                .slice(
-                    resolvedSnip.range.start.getStart(),
-                    resolvedSnip.range.end.getEnd()
-                )
-        }
-        labeledSnippets[resolvedSnip.id] = {
-            text
-        }
-    }
-    return labeledSnippets
-}
+    const openBlocks: SnipStart[] = []
 
-const extractSnippetRangesFromFile = (sourceFile: SourceFile): SnipRange[] => {
-    const snipRanges: SnipRange[] = []
-    const openBlocks: ParsedSnip[] = []
-    const comments = sourceFile.getDescendantsOfKind(
-        SyntaxKind.SingleLineCommentTrivia
-    )
-    for (const comment of comments) {
-        if (comment.getText().includes("@snip")) {
-            const parsedSnip = parseSnipComment(comment)
+    const textSplitByNewLine = sourceFileText.split("\n")
+    let text
+    for (const [i, lineText] of textSplitByNewLine.entries()) {
+        if (lineText.includes("@snip")) {
+            const parsedSnip = parseSnipComment(lineText)
             if (parsedSnip.kind === "@snipStart") {
-                openBlocks.push(parsedSnip)
+                let startSnip = parsedSnip as SnipStart
```

Comment:
Ideally it should be feasible to have types here that can actually work correctly via narrowing without having to be cast. Also, let's avoid mutating `startSnip` directly and instead maybe just destructure it to add the lineNumber.